File: Domain/Model/ValueObjects/SessionId.swift
import Foundation

/// Value object representing a unique coding session identifier
struct SessionId {
    private let value: UUID

    /// Create a new unique session ID
    init() {
        self.value = UUID()
    }

    /// Recreate from existing UUID
    init(uuid: UUID) {
        self.value = uuid
    }

    /// Get the UUID value
    var uuid: UUID {
        return value
    }

    /// String representation
    var stringValue: String {
        return value.uuidString
    }
}

// Make it Equatable
extension SessionId: Equatable {
    static func == (lhs: SessionId, rhs: SessionId) -> Bool {
        return lhs.value == rhs.value
    }
}

// Make it Hashable
extension SessionId: Hashable {
    func hash(into hasher: inout Hasher) {
        hasher.combine(value)
    }
}

// Make it Codable
extension SessionId: Codable {
    init(from decoder: Decoder) throws {
        let container = try decoder.singleValueContainer()
        value = try container.decode(UUID.self)
    }

    func encode(to encoder: Encoder) throws {
        var container = encoder.singleValueContainer()
        try container.encode(value)
    }
}

File: Domain/Model/ValueObjects/SessionId.cs
using System;

namespace InterviewMasterApp.Domain.ValueObjects
{
    /// <summary>
    /// Value object representing a unique session identifier.
    /// Migrated from Swift: SessionId.swift
    /// </summary>
    public class SessionId
    {
        public Guid Value { get; }

        /// <summary>
        /// Create a new unique session ID.
        /// </summary>
        public SessionId()
        {
            Value = Guid.NewGuid();
        }

        /// <summary>
        /// Recreate from existing GUID.
        /// </summary>
        public SessionId(Guid guid)
        {
            Value = guid;
        }

        /// <summary>
        /// String representation (UUID string format).
        /// </summary>
        public string StringValue => Value.ToString();

        public override bool Equals(object? obj)
        {
            return obj is SessionId other && Value == other.Value;
        }

        public override int GetHashCode()
        {
            return Value.GetHashCode();
        }

        public override string ToString()
        {
            return Value.ToString();
        }

        public static bool operator ==(SessionId? left, SessionId? right)
        {
            if (ReferenceEquals(left, right))
                return true;
            if (ReferenceEquals(left, null) || ReferenceEquals(right, null))
                return false;
            return left.Value == right.Value;
        }

        public static bool operator !=(SessionId? left, SessionId? right)
        {
            return !(left == right);
        }
    }
}
